extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/boot/FastBootMetadataBuilder.java

```diff
@@ -132,6 +132,14 @@ public FastBootMetadataBuilder(final PersistenceUnitDescriptor persistenceUnit,
 
         // Build the "standard" service registry
         ssrBuilder.applySettings(buildTimeSettings.getSettings());
+
+        //Hardcoded Quarkus overrides:
+        ssrBuilder.applySetting(AvailableSettings.ALLOW_ENHANCEMENT_AS_PROXY, Boolean.TRUE.toString());
+        ssrBuilder.applySetting(AvailableSettings.ORDER_UPDATES, Boolean.TRUE.toString());
+
+	//Agroal already does disable auto-commit, so ORM should trust that:
+        ssrBuilder.applySetting(AvailableSettings.CONNECTION_PROVIDER_DISABLES_AUTOCOMMIT, Boolean.TRUE.toString());
+
```

Comment:
Not a big deal but I think they should be added in `mergeSettings()` instead. That's where the others are.

Otherwise, as discussed this afternoon, it looks good.